cognite/extractorutils/unstable/core/tasks.py

```diff
@@ -1,30 +1,233 @@
-from abc import ABC
+import logging
 from collections.abc import Callable
-from dataclasses import dataclass
+from traceback import format_exception
+from typing import TYPE_CHECKING, Literal
 
-from cognite.extractorutils.unstable.configuration.models import ScheduleConfig
+from typing_extensions import assert_never
 
-__all__ = ["ScheduledTask", "ContinuousTask", "StartupTask", "Task"]
+from cognite.extractorutils.unstable.configuration.models import (
+    CronConfig,
+    IntervalConfig,
+    ScheduleConfig,
+    TimeIntervalConfig,
+)
+from cognite.extractorutils.unstable.core.errors import Error, ErrorLevel
 
+if TYPE_CHECKING:
+    from cognite.extractorutils.unstable.core.base import Extractor
 
-@dataclass
-class _Task(ABC):
-    name: str
-    target: Callable[[], None]
+__all__ = ["ScheduledTask", "ContinuousTask", "StartupTask", "Task", "TaskContext"]
+
+
+class TaskContext:
+    def __init__(self, task: "Task", extractor: "Extractor"):
+        self._task = task
+        self._extractor = extractor
+
+        self._logger = logging.getLogger(f"{self._extractor.EXTERNAL_ID}.{self._task.name.replace(' ', '')}")
+
+    def debug(self, message: str) -> None:
+        self._logger.debug(message)
+
+    def info(self, message: str) -> None:
+        self._logger.info(message)
+
+    def begin_warning(
```

Comment:
Idea: Could this be implemented as an interface-like that requires you to define `_error` and provides all the `begin_error`, etc. methods? There's some code duplication here that I think would be nice to avoid.

Reply:
that's a good idea. I was thinking about how to reuse the code here a bit, but was struggling to find some abstraction that made sense. Splitting out the `_error` method and making an abstract base class for the rest is nice, I like it.